### PR TITLE
Fix #594: External links didn't get replaced in index page

### LINKED ISSUE #594: In index page, external links didn't get replaced with spanned links.
### I agree and want to create new issue 

- [X] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [X] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [X] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 
External links should get replaced with spanned links in all pages.

### Actual behavior 
In the post page, external links got replaced. In index page, however, external links (from posts with `<!-- more -->`) didn't get replaced.

### Steps to reproduce the behavior 
1. `hexo init`
2. (IMPORTANT) In `hello-world.md`, add `<!-- more -->` after a few external links.
3. Set `theme` to `next` in hexo config.
4. Set `exturl` option to `true` in theme config.
5. Check the index page.

* Link to demo site with this issue: N/A
* Link(s) to source code or any usefull link(s): N/A

### Node.js and NPM Information

```
v10.15.0
6.4.1
```

### Package dependencies Information

```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "hexo": {
    "version": "3.8.0"
  },
  "dependencies": {
    "hexo": "^3.2.0",
    "hexo-deployer-rsync": "^0.1.3",
    "hexo-generator-archive": "^0.1.4",
    "hexo-generator-category": "^0.1.3",
    "hexo-generator-feed": "^1.2.2",
    "hexo-generator-index": "^0.2.0",
    "hexo-generator-searchdb": "^1.0.8",
    "hexo-generator-tag": "^0.2.0",
    "hexo-renderer-ejs": "^0.3.0",
    "hexo-renderer-marked": "^0.3.0",
    "hexo-renderer-stylus": "^0.3.1",
    "hexo-server": "^0.2.0"
  }
}
```

#### Hexo version

```
hexo: 3.8.0
hexo-cli: 1.1.0
os: Linux 4.9.125-linuxkit linux x64
http_parser: 2.8.0
node: 10.15.0
v8: 6.8.275.32-node.45
uv: 1.23.2
zlib: 1.2.11
ares: 1.15.0
modules: 64
nghttp2: 1.34.0
napi: 3
openssl: 1.1.0j
icu: 62.1
unicode: 11.0
cldr: 33.1
tz: 2018e
```

#### Hexo Configuration

```yml
theme: next
```

### NexT Information

**NexT Version:**

- [ ] Latest Master branch
- [X] Latest Release version
- [ ] Old version

**NexT Scheme:**

- [X] All schemes
- [ ] Muse
- [ ] Mist
- [ ] Pisces
- [ ] Gemini

#### NexT Configuration:

```yml
exturl: true
```

### Other Information 
image

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**  

- [x] Bugfix.
- [ ] Feature.
- [x] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved #594

## What is the new behavior?
Description about this pull, in several words...

* Screens with this changes: N/A
* Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: scripts/filters/exturl.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ hexo.extend.filter.register('after_post_render', function(data) {
     // If title atribute filled, set it as title; if not, set url as title.
     var title = $(this).attr('title') || href;
 
-    var encoded = new Buffer(href).toString('base64');
+    var encoded = new Buffer.from(href).toString('base64');
```
[the remark this answers is not in the document]
Comment (JiangTJ) [reply scripts/filters/exturl.js]: 😂 
Maybe not need key `new`
image

[the remark this answers is not in the document]
Comment (stevenjoezhang) [reply scripts/filters/exturl.js]: Ah, you're right... No more warnings from codacy-bot

[the remark this answers is not in the document]
Comment (JiangTJ) [reply scripts/filters/exturl.js]: Emm... it still reviewing.
image

**Comment by ivan-nginx** [comment]
What is better? We can redefine this option in [Codacy](https://app.codacy.com/project/theme-next/hexo-theme-next/dashboard).

**Pull request merged**